### PR TITLE
vendor: update module github.com/docker/distribution to v2.8.3+incomp…

### PR DESCRIPTION
**Pull request by giuseppe**
…atible

alternative to: https://github.com/containers/podman/pull/20233

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by mheon** [comment]
`go mod verify` is angry

File: pkg/domain/infra/abi/play.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ import (
 	"github.com/containers/podman/v4/pkg/util"
 	"github.com/containers/podman/v4/utils"
 	"github.com/coreos/go-systemd/v22/daemon"
-	"github.com/docker/distribution/reference"
+	"github.com/distribution/reference"
```
Comment (vrothberg) [inline pkg/domain/infra/abi/play.go:36]: Can we use `"github.com/containers/image/v5/docker/reference"` instead?  Looks like a `go-import` issue to me choosing the wrong package.

**Review by vrothberg — CHANGES_REQUESTED** [review]

File: pkg/domain/infra/abi/play.go (hunk 1 above)
Comment (giuseppe) [reply pkg/domain/infra/abi/play.go]: sure, that seems better.

Amended and pushed a new version

**Review by vrothberg — APPROVED** [review]
LGTM

File: pkg/domain/infra/abi/play.go (hunk 1 above)
Comment (vrothberg) [reply pkg/domain/infra/abi/play.go]: Thanks!

**Pull request merged**